example-viewbinding/src/main/java/com/marianatek/dkouple/viewbinding/Components.kt

```diff
@@ -0,0 +1,33 @@
+package com.marianatek.dkouple.viewbinding
+
+import android.annotation.SuppressLint
+import android.view.View
+import com.marianatek.dkouple.viewbinding.databinding.ComponentItemTextBinding
+import com.seannajera.dkouple.ComponentViewBinding
+import com.seannajera.dkouple.Component
+import com.seannajera.dkouple.DKoupleComponent
+import com.seannajera.dkouple.DKoupleView
+import com.seannajera.dkouple.FactoryView
+
+@SuppressLint("NonConstantResourceId")
```

Comment:
`nice-to-have` Gradle 8 will break the code generation feature of DKouple (they keep delaying this change so they might delay it further, but one should assume this time it will happen). This won't break the reflection based layout id lookup from the annotation at runtime to in the adapter nor will it break a non-generated factory.

I've never profiled the annotation reflection for performance, but it was necessary to make the code generation work as well as to make the api look nicer.

Now that annotations won't work for our use case, it would be nice to redesign DKouple to not use them. I'd love to explore using [KSP](https://kotlinlang.org/docs/ksp-why-ksp.html) instead which would be WAY easier to maintain than an annotation processor.

This could be a part of a larger/separate issue.